integration/hive/src/main/java/org/apache/carbondata/hive/CarbonHiveRecordReader.java

```diff
@@ -111,58 +108,46 @@ private void initialize(InputSplit inputSplit, Configuration conf) throws IOExce
     } else {
       columnTypes = TypeInfoUtils.getTypeInfosFromTypeString(columnTypeProperty);
     }
+
+    if (valueObj == null) {
+      valueObj = new ArrayWritable(Writable.class, new Writable[columnTypes.size()]);
+    }
```

Comment:
columnTypes.size() and queryModel.getProjectionColumns().length is different, why need to do this change ?